sdk/dotnet/Pulumi/X/Automation/StackSettings.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+
+namespace Pulumi.X.Automation
+{
+    public class StackSettings
+    {
+        /// <summary>
+        /// This stack's secrets provider.
+        /// </summary>
+        public string? SecretsProvider { get; set; }
+
+        /// <summary>
+        /// This is the KMS-encrypted ciphertext for the data key used for secrets
+        /// encryption. Only used for cloud-based secrets providers.
+        /// </summary>
+        public string? EncryptedKey { get; set; }
+
+        /// <summary>
+        /// This is this stack's base64 encoded encryption salt. Only used for
+        /// passphrase-based secrets providers.
+        /// </summary>
+        public string? EncryptionSalt { get; set; }
+
+        /// <summary>
+        /// This is an optional configuration bag.
+        /// </summary>
+        public IDictionary<string, string>? Config { get; set; } // TODO: figure out what to do for value type argument here
```

Comment:
Any thoughts on how to represent the value type argument of this dictionary here, since there is some union type representation going on in the TS implementation? Should we use the `Union<Tx, Ty>` implementation I saw in the repo or can you think of a better way?

Reply:
[Here](https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/x/automation/stackSettings.ts) is the TS implementation.

Reply:
For a little more context, here's the reference to the original format: https://github.com/pulumi/pulumi/blob/6d09fe32df2ac7eca3db08c3b4a61dddcb7d756e/sdk/go/common/resource/config/value.go#L25

Serde is a little complicated on this type. In practice, I think it will be rare for users to manipulate config via this interface (any secret values retrieved from this will be encrypted).